### DISCUSSION TITLE
¿Consulta sobre acceso a data base?

### DISCUSSION THREAD

**Discussion by NicolasPLG** [Cotalker/documentation#118] [2. Questions and Answers] [answered]
Hola tengo la siguiente consulta

Estoy configurando un rol para que el usuario pueda ver y editar en data base. El tema es el siguiente

Con los permisos db-properties-write, db-properties-read. El tema es el siguiente: No puedo editar la información y no la puedo visualizar.

Agregué el web-admin-read, y me funcionaba de manera correcta, pero con la nueva versión web, me está mostrando todos los campos del administrador. En otrs implementaciones tenía separado como user y property (con read y write respectivamente), pero no podía editar y tenía que agregar el web-admin-read para que se viera sol esos campos.

![image](https://user-images.githubusercontent.com/61419471/116447570-2224e900-a826-11eb-9912-cb373ad28ae8.png)

Al percatarme de la consola me arroja esto en la consola

![image](https://user-images.githubusercontent.com/61419471/116447334-deca7a80-a825-11eb-88b3-6923a858708a.png)

El segundo error puede ser porque quizás esté usando los extratype 2.0 (agregados desde la base) y que ahora no estén funcionando (pero con el rol web-admin-read si puedo ver la información.

Pero con el error 1 del knowledgebase este cliente no tiene instalado KB, creo que en otra cía tambien tenía el error.

Gracias por la ayuda y espero su respuesta

**Comment by acedward** [comment]
Tienes `admin-properties-write` y `admin-properties-write` ?
Puede que sean necesarios.

**Reply by NicolasPLG** [reply]
Ya los he probado y no me funciona me queda lopeando. Me funciona al colocar el web-admin-read y actualizo la cuenta de usuario. Sin embargo, si cierro la sesión y vuevlo a ingresar, me aparece todas las opciones del administrador :(. Quizas se rompió algo?. @gmoplaza nos puede ayudar :>

**Comment by gmoplaza** [comment, marked as answer]
@NicolasPLG no logro entender tu problema. Voy a detallar varias cosas para ver si algo te puede ayudar.

- Los permisos tienen dos terminaciones (los nuevos que siguen el formato (admin/db/report)-seccion-(write/read))  _write_ o _read_. 
1. Permisos terminados en _read_, **solo** pueden entrar a la sección y **no** editar.
2. Permisos terminados en _write_, dan permiso de escritura y por lo tanto también de lectura. Por lo que **no** es necesario agregar _read_ .

- La sección **Database** solo muestra **elementos** (properties), por lo que los permisos involucrados deberían si o si ser de la forma *-properties-*.

- Se está buscando mejorar toda la complejidad de los permisos. La @ccarrascor y @acedward han trabajado en una librería que funciona de la siguiente forma.
![image](https://user-images.githubusercontent.com/9313324/116583706-a2aa1f00-a8e4-11eb-913f-9830f878309f.png)
Por lo que el permiso db-properties-write debería ser suficiente.

- Permisos legacy: Estos deberían ser incorporado también a la librería que mencioné en el punto anterior, pero puede ser que aún no esté listo. De todas formas no entiendí si al agregar **web-admin-read** te funcionó o no.

1. En caso que te haya funcionado y hayas encontrado otro problema, por favor inicia otro issue ya que este tiene el tema: _¿Consulta sobre acceso a data base?_ y tu mismo respondiste tu problema.
2. Si aún no funciona puedes explicar exactamente que permisos tienes y que es lo que estás buscando lograr, pero nada más que eso y con gusto veo que es lo que te está pasando.

**Reply by NicolasPLG** [reply]
hola @gmoplaza gracias por la respuesta e info, lo que quiero hacer es a través de data base, poder ver la información de las propiedades y/o editarlas (se está usando extratype 2.0 (antiguos).

Adjunto link con el video y creo que puedes ver bien cuál es el problema: https://drive.google.com/file/d/1WXJD6J9LF7FPej5QS3Tv1Vc2RbY3U4tS/view?usp=sharing
